stan/math/rev/scal/fun/frexp.hpp

```diff
@@ -0,0 +1,39 @@
+#ifndef STAN_MATH_REV_SCAL_FUN_FREXP_HPP
+#define STAN_MATH_REV_SCAL_FUN_FREXP_HPP
+
+#include <stan/math/rev/core.hpp>
+#include <cmath>
+#include <limits>
+
+namespace stan {
+  namespace math {
+
+      class frexp_vari : public op_v_vari {
+      public:
+          explicit frexp_vari(vari* avi, int* b) :
+          op_v_vari(std::frexp(avi->val_, b), avi) {
+          }
+          void chain() {
+              if (unlikely(boost::math::isnan(avi_->val_)))
+                  avi_->adj_ = std::numeric_limits<double>::quiet_NaN();
```

Comment:
Just put some doc in here that the derivative is zero for `frexp`.  

We'd really like all of the classes and constructors getting added now to be documented.  We were very lax before and have been trying to catch up in files we touch, too.  It doesn't have to be extensive.